src/x_response_time.rs

```diff
@@ -0,0 +1,59 @@
+// Borrowed code from
+// https://api.rocket.rs/v0.4/rocket/fairing/trait.Fairing.html#request-local-state
+
+use std::time::SystemTime;
+
+use rocket::fairing::{Fairing, Info, Kind};
+use rocket::http::Status;
+use rocket::request::{self, FromRequest};
+use rocket::{Data, Outcome, Request, Response};
+
+/// Fairing for timing requests.
+pub struct RequestTimer;
+
+/// Value stored in request-local state.
+#[derive(Copy, Clone)]
+struct TimerStart(Option<SystemTime>);
+
+impl Fairing for RequestTimer {
+    fn info(&self) -> Info {
+        Info {
+            name: "Request Timer",
+            kind: Kind::Request | Kind::Response,
+        }
+    }
+
+    /// Stores the start time of the request in request-local state.
+    fn on_request(&self, request: &mut Request, _: &Data) {
+        // Store a `TimerStart` instead of directly storing a `SystemTime`
+        // to ensure that this usage doesn't conflict with anything else
+        // that might store a `SystemTime` in request-local cache.
+        request.local_cache(|| TimerStart(Some(SystemTime::now())));
+    }
+
+    /// Adds a header to the response indicating how long the server took to
+    /// process the request.
+    fn on_response(&self, request: &Request, response: &mut Response) {
+        let start_time = request.local_cache(|| TimerStart(None));
+        if let Some(Ok(duration)) = start_time.0.map(|st| st.elapsed()) {
+            let ms = duration.as_secs() * 1000 + duration.subsec_millis() as u64;
+            response.set_raw_header("X-Response-Time", format!("{} ms", ms));
+        }
+    }
+}
+
+/// Request guard used to retrieve the start time of a request.
```

Comment:
I _think_ we do not need anything from this line onwards. Unless you plan to use the timer on a route function.

Reply:
Ok! I will remove it.